File: app/src/main/java/dev/suhockii/lifetest/util/AppUtils.java
package dev.suhockii.lifetest.util;

import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.ProgressBar;

import com.bumptech.glide.load.DataSource;
import com.bumptech.glide.load.engine.GlideException;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.target.Target;

public class AppUtils {
    @NonNull
    public static RequestListener<Drawable> getGlideListener(ProgressBar progressBar) {
        return new RequestListener<Drawable>() {

            @Override
            public boolean onLoadFailed(@Nullable GlideException e, Object model,
                                        Target<Drawable> target, boolean isFirstResource) {
                return false;
            }

            @Override
            public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target,
                                           DataSource dataSource, boolean isFirstResource) {
                if (progressBar != null) {
                    progressBar.setVisibility(View.INVISIBLE);
                }
                return false;
            }
        };
    }
}

File: app/src/main/java/dev/suhockii/lifetest/util/AppUtils.kt
package dev.suhockii.lifetest.util

import android.graphics.drawable.Drawable
import android.view.View
import android.widget.ProgressBar

import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target

object AppUtils {
    fun getGlideListener(progressBar: ProgressBar?): RequestListener<Drawable> {
        return object : RequestListener<Drawable> {

            override fun onLoadFailed(
                e: GlideException?, model: Any,
                target: Target<Drawable>, isFirstResource: Boolean
            ): Boolean {
                return false
            }

            override fun onResourceReady(
                resource: Drawable, model: Any, target: Target<Drawable>,
                dataSource: DataSource, isFirstResource: Boolean
            ): Boolean {
                if (progressBar != null) {
                    progressBar.visibility = View.INVISIBLE
                }
                return false
            }
        }
    }
}
